### ISSUE TITLE
[android] [in_app_purchase] introductoryPriceMicros no value

### ISSUE THREAD

**Issue by bohowu** [flutter/flutter#90244]
## Steps to Reproduce

1. Install `in_app_purchase `.
2. Configure to work on android
3. Try to get introductoryPriceMicros

**Expected results:** GooglePlayProductDetails.skuDetails.introductoryPriceMicros should have value

**Actual results:** GooglePlayProductDetails.skuDetails.introductoryPriceMicros is empty

<details>
  <summary>More Info</summary>

https://github.com/flutter/plugins/blob/701fdb6eab07c99d83fe583f1b7630100fe01a0e/packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/sku_details_wrapper.dart#L71

sku_details_wrapper.dart introductoryPriceMicros JsonKey should be 'introductoryPriceAmountMicros' to parse the correct value

</details>

**Comment by dodatw** [comment]
I face the same issue, it seems a typo bug?

**Comment by darshankawar** [comment]
Observed the same on master version of the plugin.

https://github.com/flutter/plugins/blob/701fdb6eab07c99d83fe583f1b7630100fe01a0e/packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/sku_details_wrapper.dart#L71

**Comment by ThrowJojo** [comment]
I don't think this is completely fixed. I think the change may have introduced a new bug because when I try to use the latest `in_app_purchase` (or more specifically `in_app_purchase_android`, 0.1.4+7) I'm seeing this error in the debug console:

```
I/flutter (13793): ----------------FIREBASE CRASHLYTICS----------------
I/flutter (13793): type 'int' is not a subtype of type 'String?' in type cast
I/flutter (13793): #0      _$SkuDetailsWrapperFromJson (package:in_app_purchase_android/src/billing_client_wrappers/sku_details_wrapper.g.dart:15:47)
I/flutter (13793): #1      new SkuDetailsWrapper.fromJson (package:in_app_purchase_android/src/billing_client_wrappers/sku_details_wrapper.dart:56:7)
I/flutter (13793): #2      _$SkuDetailsResponseWrapperFromJson.<anonymous closure> (package:in_app_purchase_android/src/billing_client_wrappers/sku_details_wrapper.g.dart:59:35)
I/flutter (13793): #3      MappedListIterable.elementAt (dart:_internal/iterable.dart:412:31)
I/flutter (13793): #4      ListIterator.moveNext (dart:_internal/iterable.dart:341:26)
I/flutter (13793): #5      new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:188:27)
I/flutter (13793): #6      new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
I/flutter (13793): #7      new List.of (dart:core-patch/array_patch.dart:50:28)
I/flutter (13793): #8      ListIterable.toList (dart:_internal/iterable.dart:212:44)
I/flutter (13793): #9      _$SkuDetailsResponseWrapperFromJson (package:in_app_purchase_android/src/billing_client_wrappers/sku_details_wrapper.g.dart:60:14)
I/flutter (13793): #10     new SkuDetai
I/flutter (13793): ----------------------------------------------------
```

It's pointing to this line:

```
package:in_app_purchase_android/src/billing_client_wrappers/sku_details_wrapper.g.dart:15:47
```

This could potentially be because I haven't set any introductory price for my SKUs, so the API is returning an int value(0) for `introductoryPriceAmountMicros` instead of a String.

**Comment by ThrowJojo** [comment]
By the way, everything seems to work fine if I pin `in_app_purchase_android` to 0.1.4+6 in my pubspec.yaml.

**Comment by dodatw** [comment]
Is there a new in_app_purchase include this fixing ? I use in_app_purchase for android and iOS.

**Comment by mvanbeusekom** [comment]
@ThrowJojo I have created a new [PR](https://github.com/flutter/plugins/pull/4370) to resolve this problem. Apologies for the inconvenience. 

@dodatw the best way to access this fix is to add a dependency on the `in_app_purchase_android` package in your pubspec.yaml file. However is you depend on the `in_app_purchase` plugin you should be good by simply running `flutter pub upgrade` to upgrade all dependencies. Note the the current version (0.1.4+7) of the `in_app_purchase_android` contains the above bug described by @ThrowJojo. So until [PR](https://github.com/flutter/plugins/pull/4370) is merged it is better to add a dependency override and stick to version 0.1.4+6:

```yaml
dependency_overrides:
  in_app_purchase_android: '0.1.4+6'
```

 (remove the dependency override and run `flutter pub upgrade` when [PR](https://github.com/flutter/plugins/pull/4370) is merged and version 0.1.4+8 is published).
